lib/runtime/interface.go

```diff
@@ -16,7 +16,6 @@ import (
 // Instance is the interface a v0.8 runtime instance must implement
 type Instance interface {
 	UpdateRuntimeCode([]byte) error
-	CheckRuntimeVersion([]byte) (Version, error)
```

Comment:
I understand why you removed this function from the interface since it creates a new instance to do the version check, but how will we mock this out now? We'd have to change the function signatures of all callers of this function to include a `func([]byte) (Version, error)`

Reply:
I don't think it's fast at all.  It's instantiating a wasmer instance, loading the code in, binding the imports, etc.  Now all of the unit tests are runtime specific if we remove it from the `Instance` interface.  

Why aren't we just implementing a `RuntimeVersion() (Version, error)` (no blob) on `Instance`.  Then anyone who needs to version check needs to instantiate an instance via `NewInstance`.

Reply:
It takes 500ms to load the version, so not that fast indeed.
**But** if you have a look, there was zero place we were configuring the `CheckRuntimeVersion` mock expectation, so it won't change a thing with our code as it is. And as I mentioned above, if in the future we want to mock the runtime check (for test coverage or testing speed), we can just set it as a field on the receiving struct (like we would with `timeNow func() time.Time` to mock time).

Reply:
I looked at the interface again.  The `Instance` interface includes a `Version() (Version, error)` method.  I think we should remove `CheckRuntimeVersion` from the runtime package and make this a private method in `dot/state` and a helper function for the tests. What do you think?

Reply:
Good point, although that would require to export a bunch of things such as `setupVM`, the wasmer `Instance` struct fields `vm` and `ctx`, and I think ultimately we should keep this in the wasmer package since it's quite tightly coupled with wasmer thingies? I would rather export `CheckRuntimeVersion` than these other function/fields, what do you think?

Reply:
why do you need to export those private functions? To check a version of a given wasm blob, you instantiate an instance by calling `NewInstance(code []byte, cfg runtime.InstanceConfig)` and call `Version()` on that instance.

Reply:
Good point, sorry I misunderstood.
I changed `CheckRuntimeVersion` to use `NewInstance`, simplifying it a bit / reducing code duplication (although it sets up extra stuff it doesn't need, but that's a dep injection problem for another day).

On the other hand, `ext_misc_runtime_version_version_1` in `wasmer/imports.go` uses `CheckRuntimeVersion` so I think we should still keep it in the wasmer package? 🤔

Reply:
I have dissonance with having a function on the public package API that achieves the same functionality that can be achieved by instantiating an instance and calling a receiver method on that instance.  But that would lead to code duplication cause we use it in various places.  I gave it a shot in this [commit](https://github.com/ChainSafe/gossamer/commit/29e178ea16ce1caee7fbc775b9549172f315d90e), and I'm not really happy with it.  So I'm ok with leaving with leaving it in the `wasmer` package.